pkg/kine/server/watch.go

```diff
@@ -4,49 +4,78 @@ import (
 	"context"
 	"sync"
 	"sync/atomic"
+	"time"
 
 	"github.com/sirupsen/logrus"
 	"go.etcd.io/etcd/api/v3/etcdserverpb"
 	"go.etcd.io/etcd/api/v3/mvccpb"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"golang.org/x/exp/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 var (
 	watchID int64
 )
 
+// explicit interface check
+var _ etcdserverpb.WatchServer = (*KVServerBridge)(nil)
+
+// getProgressReportInterval returns the configured progress report interval, with some jitter
+func (s *KVServerBridge) getProgressReportInterval() time.Duration {
+	// add rand(1/10*notifyInterval) as jitter so that kine will not
+	// send progress notifications to watchers at the same time even when watchers
+	// are created at the same time.
+	jitter := time.Duration(rand.Int63n(int64(s.limited.notifyInterval) / 10))
```

Comment:
Given that a polling starts when a watch arrives, do we really need a jitter? I think that requests are random enough not to care.

Reply:
I believe this could make sense for large deployments. If we create a deployment of an application with say 10,000 pods, watchers are started at the same time.

Reply:
From discussion on MM, removing the jitter until we find evidence for its need.